test/libsolidity/semanticTests/externalContracts/snark.sol

```diff
@@ -35,34 +35,26 @@ library Pairing {
 
 	/// @return r the sum of two points of G1
 	function add(G1Point memory p1, G1Point memory p2) internal returns (G1Point memory r) {
-		uint[4] memory input;
+		uint[6] memory input;
 		input[0] = p1.X;
 		input[1] = p1.Y;
 		input[2] = p2.X;
 		input[3] = p2.Y;
-		bool success;
-		assembly {
-			success := call(sub(gas(), 2000), 6, 0, input, 0xc0, r, 0x60)
-			// Use "invalid" to make gas estimation work
-			switch success case 0 { invalid() }
-		}
+		(bool success, bytes memory encodedResult) = address(6).call(abi.encode(input));
```

Comment:
This is something I don't really understand. According to https://www.evm.codes/precompiled `ecadd` (precompile at address `0x06`) accepts 4 uints. However the original code was passing in 6 of them (`0xc0` bytes). It reverts if I try to pass in 4.

Similarly, `ecmul` below seems to expect 4 rather than 3 parameters.

Reply:
BTW, I'm the reason I'm removing the assembly blocks in this file is of course that `call()` is only available on legacy and the equivalent `extcall()` only on EOF. We currently can't have assembly that works on both, but we can use `<address>.call()` in Solidity, which gets automatically translated to the right opcode.

This contract seems to be using assembly only to work around some problem related to gas estimation (and maybe also to make the call cheaper), which is not an issue in our tests.

Reply:
@ekpyron got to the bottom of it:

> My guess would be that it's a typo in ancient versions of this (there's very old snippets of this piece of code around) (i.e. it should always have been 0x80 as size, but the oldest versions dating back to 2017 or older just had 0xC0, and since that also works as per spec, nobody ever fixed it, and instead the hard-coded cases in our implementation were adjusted to accommodate those - funny actually :-)) (see https://github.com/Zokrates/ZoKrates/issues/1364 and such - and https://github.com/Zokrates/ZoKrates/issues/1364#issuecomment-2017827015 in particular :-))

#15788 fixes this.